In [1]:
import pandas as pd
import numpy as np

In [2]:
df = pd.read_csv('chennai_reviews_2.csv')

In [3]:
from sklearn.feature_extraction.text import CountVectorizer,TfidfTransformer
from sklearn.model_selection import train_test_split

In [4]:
X = df['Review_Text']
y = df['Sentiment']

In [7]:
X_train,X_test,y_train,y_test = train_test_split(X,y,test_size=0.30,random_state=42)

In [29]:
vect = CountVectorizer(stop_words='english')
tf_idf = TfidfTransformer()

In [30]:
X_train_count_vect = vect.fit_transform(X_train)
X_test_count_vect = vect.fit_transform(X_test)

In [31]:
X_train_tf_idf = tf_idf.fit_transform(X_train_count_vect)
X_test_tf_idf = tf_idf.fit_transform(X_test_count_vect)

In [14]:
from sklearn.naive_bayes import ComplementNB
from sklearn.model_selection import cross_val_score

In [25]:
cnb_clf = ComplementNB()
cnb_scores_5 = cross_val_score(cnb_clf,X_train_tf_idf,y_train,scoring='accuracy',cv=5)
cnb_scores_10 = cross_val_score(cnb_clf,X_train_tf_idf,y_train,scoring='accuracy',cv=10)
print('CV 5 Mean Accuracy for Complement NB: {}'.format(round(cnb_scores_5.mean(),2)))
print('\n')
print('CV 10 Mean Accuracy for Complement NB: {}'.format(round(cnb_scores_10.mean(),2)))

CV 5 Mean Accuracy for Complement NB: 0.78


CV 10 Mean Accuracy for Complement NB: 0.79


Without Stop Words:

CV 5 Mean Accuracy for Complement NB: 0.79

CV 10 Mean Accuracy for Complement NB: 0.78

With Stop Words:

CV 5 Mean Accuracy for Complement NB: 0.78


CV 10 Mean Accuracy for Complement NB: 0.79

In [27]:
from sklearn.svm import SVC

In [32]:
svc_clf = SVC(gamma='scale',kernel='linear')
svc_scores_5 = cross_val_score(svc_clf,X_train_tf_idf,y_train,scoring='accuracy',cv=5)
svc_scores_10 = cross_val_score(svc_clf,X_train_tf_idf,y_train,scoring='accuracy',cv=10)

print('CV 5 Mean Accuracy for Linear SVC: {}'.format(round(svc_scores_5.mean(),2)))
print('\n')
print('CV 10 Mean Accuracy for Linear SVC: {}'.format(round(svc_scores_10.mean(),2)))

CV 5 Mean Accuracy for Linear SVC: 0.82


CV 10 Mean Accuracy for Linear SVC: 0.82


With Stop Words:

CV 5 Mean Accuracy for Linear SVC: 0.83

CV 10 Mean Accuracy for Linear SVC: 0.83

Without Stop Words:

CV 5 Mean Accuracy for Linear SVC: 0.82


CV 10 Mean Accuracy for Linear SVC: 0.82